src/Sentry/SentryMonitorOptions.cs

```diff
@@ -0,0 +1,188 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry;
+
+internal enum SentryMonitorScheduleType
+{
+    None,
+    Crontab,
+    Interval
+}
+
+/// <summary>
+/// Sentry's intervals for monitors
+/// </summary>
+public enum SentryMonitorInterval
+{
+    /// <summary>
+    /// Year
+    /// </summary>
+    Year,
+
+    /// <summary>
+    /// Month
+    /// </summary>
+    Month,
+
+    /// <summary>
+    /// Week
+    /// </summary>
+    Week,
+
+    /// <summary>
+    /// Day
+    /// </summary>
+    Day,
+
+    /// <summary>
+    /// Hour
+    /// </summary>
+    Hour,
+
+    /// <summary>
+    /// Minute
+    /// </summary>
+    Minute
+}
+
+/// <summary>
+/// Sentry's options for monitors
+/// </summary>
+public class SentryMonitorOptions : ISentryJsonSerializable
+{
+    private SentryMonitorScheduleType _type = SentryMonitorScheduleType.None;
+    private string? _crontab;
+    private int? _interval;
+    private SentryMonitorInterval? _unit;
+
+    // Breakdown of the validation
+    // ^(\*|([0-5]?\d))                 Minute  0 - 59
+    // (\s+)(\*|([01]?\d|2[0-3]))       Hour    0 - 23
+    // (\s+)(\*|([1-9]|[12]\d|3[01]))   Day     1 - 31
+    // (\s+)(\*|([1-9]|1[0-2]))         Month   1 - 12
+    // (\s+)(\*|([0-7]))                Weekday 0 - 7
+    // $                                End of string
+    private readonly Regex _crontabValidation = new(
+        @"^(\*|([0-5]?\d))(\s+)(\*|([01]?\d|2[0-3]))(\s+)(\*|([1-9]|[12]\d|3[01]))(\s+)(\*|([1-9]|1[0-2]))(\s+)(\*|([0-7]))$",
+        RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="cronTab"></param>
+    public void Interval(string cronTab)
+    {
+        if (_type is not SentryMonitorScheduleType.None)
+        {
+            throw new ArgumentException("You tried to set the interval twice. The Check-Ins interval is supposed to be set only once.");
+        }
+
+        if (!_crontabValidation.IsMatch(cronTab))
+        {
+            throw new ArgumentException("The provided crontab does not match the expected format of '* * * * *' " +
+                                        "translating to 'minute', 'hour', 'day of the month', 'month', and 'day of the week'.");
+        }
+
+        _type = SentryMonitorScheduleType.Crontab;
+        _crontab = cronTab;
+    }
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="interval"></param>
+    /// <param name="unit"></param>
+    public void Interval(int interval, SentryMonitorInterval unit)
```

Comment:
Given we have this method, is there any need for the more fragile overload that allows a `string cronTab` argument? Could we just have this more robust way to create intervals (guaranteed by compile time static analysis)?

Reply:
I'd like to avoid having to reimplement something that lets me run and expect a job every Thursday when crontabs are a somewhat standardized way to set these up.